components/script/stylesheet_loader.rs

```diff
@@ -167,18 +167,23 @@ impl FetchResponseListener for StylesheetContext {
             successful = metadata.status.map_or(false, |(code, _)| code == 200);
         }
 
+        let url = self.source.url();
         let owner = elem.upcast::<Element>().as_stylesheet_owner()
             .expect("Stylesheet not loaded by <style> or <link> element!");
+
+        document.finish_load(LoadType::Stylesheet(url));
         if owner.parser_inserted() {
             document.decrement_script_blocking_stylesheet_count();
         }
 
-        let url = self.source.url();
-        document.finish_load(LoadType::Stylesheet(url));
-
         if let Some(any_failed) = owner.load_finished(successful) {
             let event = if any_failed { atom!("error") } else { atom!("load") };
             elem.upcast::<EventTarget>().fire_event(event);
+
+            if owner.parser_inserted() {
+                document.process_pending_parsing_blocking_script_if_any();
+            }
+            document.maybe_fire_load_event();
```

Comment:
This code is a counterexample to my earlier comment, but I don't believe it would be a problem to move the call to maybe_fire_load_event out of the conditional block, which would make my idea work.